commit 277de8ac3017916f2d09b3d7059402d483d22dcc
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon Dec 11 09:18:32 2017 +0000

    ARM: Activate FIQs to avoid __irq_startup warnings
    
    There is a new test in __irq_startup that the IRQ is activated, which
    hasn't been the case for FIQs since they bypass some of the usual setup.
    
    Augment enable_fiq to include a call to irq_activate to avoid the
    warning.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/kernel/fiq.c b/arch/arm/kernel/fiq.c
index cd1234c103fc..a0f96e3b8898 100644
--- a/arch/arm/kernel/fiq.c
+++ b/arch/arm/kernel/fiq.c
@@ -56,6 +56,8 @@
 static unsigned long dfl_fiq_insn;
 static struct pt_regs dfl_fiq_regs;
 
+extern int irq_activate(struct irq_desc *desc);
+
 /* Default reacquire function
  * - we always relinquish FIQ control
  * - we always reacquire FIQ control
@@ -140,6 +142,8 @@ static int fiq_start;
 
 void enable_fiq(int fiq)
 {
+	struct irq_desc *desc = irq_to_desc(fiq + fiq_start);
+	irq_activate(desc);
 	enable_irq(fiq + fiq_start);
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1119_277de8ac3017916f2d09b3d7059402d483d22dcc.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1119_277de8ac3017916f2d09b3d7059402d483d22dcc.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
